src/clr-angular/forms/combobox/_combobox.clarity.scss

```diff
@@ -1,69 +1,188 @@
-// Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+// Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 // This software is released under MIT license.
 // The full license information can be found in LICENSE in the root directory of this project.
 
-@import '../../popover/dropdown/menu-mixins.clarity';
 @import './properties.combobox';
 
 @include exports('combobox.clarity') {
-  .clr-combobox {
-    display: inline-block;
+  // needed for the proper calculation of popover offset that we do to keep label always visible
+  .clr-combobox-form-control {
+    align-items: stretch;
   }
 
-  .clr-combobox-input-container {
-    display: flex;
-    align-items: center;
-  }
+  .clr-combobox-container {
+    position: relative;
 
-  .clr-combobox-input {
     //Dimensions
     padding: 0 $clr-ng-select-trigger-right-padding 0 $clr-ng-select-trigger-left-padding;
     min-height: $clr-ng-select-min-height;
-    width: $clr-ng-select-min-width;
+    min-width: $clr-ng-select-min-width;
     @include css-var(font-size, clr-ng-select-font-size, $clr-ng-select-font-size, $clr-use-custom-properties);
 
+    &.multi {
+      min-width: $clr-ng-multiselect-min-width;
+      padding-bottom: $clr_baselineRem_0_125;
+    }
+
     //Mixins
-    @include form-fields-appearance();
-    @include input-border-bottom-animation();
+    @include form-fields-appearance($clr-forms-border-color);
+
+    // override the above mixin for combobox-specific styles
+    @include css-var(
+      background-color,
+      clr-ng-select-input-background-color,
+      $clr-ng-select-input-background-color,
+      $clr-use-custom-properties
+    );
+
+    border-bottom: $clr-global-borderwidth solid $clr-forms-border-color;
+
+    // the border color is on the component; we don't want it on the input itself
+    .clr-input.clr-combobox-input:focus {
+      background: none;
+    }
+
+    // ideally I want to just use &:focus-within, but cross-browser support is limited
+    // for now, we programmatically add/remove .clr-focus class to achieve this
+    &:focus-within,
+    &.clr-focus {
+      border-bottom: $clr_baselineRem_2px solid $clr-forms-focused-color;
+    }
+
+    &.invalid,
+    &.invalid:focus,
+    &.clr-focus.invalid {
+      border-bottom-color: $clr-forms-invalid-color;
+    }
 
     //Others
-    outline: none;
+    .clr-combobox-input {
+      background: none;
+      border-bottom: none;
+    }
+
+    .clr-combobox-remove-btn {
+      background: transparent;
+      border: none;
+      cursor: pointer;
+      padding: $clr_baselineRem_4px $clr_baselineRem_4px;
+      color: $clr-ng-select-pill-font-color;
+    }
+
+    .label-combobox-pill {
```

Comment:
I wonder if we can make this generic and allow labels to have this outside of the combobox as well?